### PR TITLE
Implements OverlayClass.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/engine/overlay.cpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
```
Comment (OmniBlade) [inline src/game/engine/overlay.cpp:6]: Can we have a proper brief here and in the header, something like "Class for handling placement of overlay." I think I've let this slide too much.

File: src/game/engine/overlay.cpp [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+#include "iomap.h"
+#include "globals.h"
+
+#ifndef GAME_DLL
+TFixedIHeapClass<OverlayClass> g_Overlays;
+#endif
+
+HousesType OverlayClass::s_ToOwn = HOUSES_NONE;
+
+/**
```
Comment (OmniBlade) [inline src/game/engine/overlay.cpp]: Don't bother with comment blocks for ctors and operators, its clear what they do and they don't need commenting for the most part.

File: src/game/engine/overlay.cpp [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+ *
+ *
+ */
+BOOL OverlayClass::Mark(MarkType mark)
+{
+    if (!ObjectClass::Mark(mark) || mark != MARK_PUT) {
+        return false;
+    }
```
Comment (OmniBlade) [inline src/game/engine/overlay.cpp:91]: Inconsistent white space, you put a new line in before/after some scopes but not all in the same function! Me nad CCHyper almost always have whitespace where scopes open and close in the same indentation level and don't where there is a chance in indentation.

File: src/game/engine/overlay.cpp [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+                if (type->What_Type() == OVERLAY_STEEL_CRATE || type->What_Type() == OVERLAY_WOOD_CRATE) {
+                    cell_clear = cell->Is_Clear_To_Move(SPEED_TRACK);
+                } else {
+                    cell_clear = cell->Is_Clear_To_Move(SPEED_TRACK, true, true);
+                }
+            }
+        } else {
+            cell_clear = true;
+        }
+        if ((ScenarioInit || cell->Get_Overlay() == OVERLAY_NONE) && cell_clear) {
```
Comment (OmniBlade) [inline src/game/engine/overlay.cpp:128]: No new line here for example where in other scope changes there are.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by OmniBlade — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**